### ISSUE TITLE
Repr value for nan

### ISSUE THREAD

**Issue by DimitrisJim** [RustPython/RustPython#3034]
In Python:

```python
>>> repr(float('nan'))
'nan'
```

while in RustPython:

```python
>>>>> repr(float('nan'))
'NaN'
```

slight incompatibility but it is actually tested for in `cmath`.

**Comment by ShadowJonathan** [comment]
(TIL NaN is lowercase in repr in python)

**Comment by DimitrisJim** [comment]
C also displays it as `'nan'` if I'm not mistaken. I always remember seeing `'NaN'` in Javascript tbh and Rust seems to follow that. Either way, `'NaN'` always seemed weird to me, either capitalize em all or don't, why is `a` being left out like that? No respect for poor `a`.

**Comment by Codemonk-adi** [comment]
@DimitrisJim I would like to work on it
would you pls guide me a bit?

**Comment by DimitrisJim** [comment]
Yes! Take a look at float's `repr` function. This calls another function to handle formatting for the different values that might be passed in. Take a look there and try and find out which case `Nan/Inf` fall under and what you'd need to do.

**Comment by Codemonk-adi** [comment]
RustPython/Lib/reprlib.py
```
s = builtins.repr(x)
# Bugs in x.__repr__() can cause arbitrary
# exceptions -- then make up something

```
repr(float('nan'))
returns the builtin repr function called with float('nan')

I am stuck here...

**Comment by DimitrisJim** [comment]
Hey @Codemonk-adi, not sure what you mean here. Are you looking inside the implementation of `repr` for `PyFloat`? You can always fire up a debugger and get through stepping your way until the end (it takes some getting used to but its worth it).

**Comment by Codemonk-adi** [comment]
@DimitrisJim  I am sorrry to bother you again, but I am unable to use the debugger, when I run the debug option on vscode it opens an rust python interpreter, and I do enter debugging mode in vscode but the buttons are disabled.

**Comment by DimitrisJim** [comment]
@Codemonk-adi hop on over to [gitter](https://gitter.im/rustpython/Lobby) if you can. This way I can help you further w/o notifying watchers.

**Comment by DimitrisJim** [comment]
closed with #3103
